From af6e2aa5db9a539ac3755f9695e783d232ab546a Mon Sep 17 00:00:00 2001
From: wanlong <long.wan@spacemit.com>
Date: Tue, 21 May 2024 15:11:55 +0800
Subject: [PATCH 0712/1448] wireless: rtl8852bs: avoid NULL pointer dereference
 when phl_sta is NULL

[  523.051911] Unable to handle kernel NULL pointer dereference at virtual address 000000000000002c
[  523.081183] Hardware name: spacemit k1-x MUSE-Pi board (DT)
[  523.088746] epc : rtw_init_self_stainfo+0xa4/0x118 [8852bs]
[  523.099285]  ra : rtw_init_self_stainfo+0xe2/0x118 [8852bs]
[  523.233724] [<ffffffff021a23de>] _connect_disconnect_hw+0x8a/0xfc [8852bs]
[  523.245869] [<ffffffff021a4258>] _connect_msg_hdlr+0x920/0xaf0 [8852bs]
[  523.257765] [<ffffffff02259762>] cur_req_hdl+0x32/0x3c [8852bs]
[  523.268950] [<ffffffff0225a0c0>] msg_dispatch+0x238/0x3e4 [8852bs]
[  523.280393] [<ffffffff0225d7fc>] dispr_thread_loop_hdl+0x25e/0x2d4 [8852bs]
[  523.292650] [<ffffffff0225d882>] dispr_share_thread_loop_hdl+0x10/0x18 [8852bs]
[  523.305296] [<ffffffff0225744c>] $xrv64i2p1_m2p0_a2p1_c2p0_zicbom1p0_zicbop1p0_zicboz1p0_zicsr2p0_zifencei2p0_zihintpause2p0_zmmul1p0+0xa0/0x194 [8852bs]
[  523.324515] [<ffffffff8003120c>] kthread+0xb8/0xdc
[  523.331760] [<ffffffff80003bf0>] ret_from_exception+0x0/0x16

Change-Id: Ib500c73c9de21bc13186c65c4059d27a52b819f4
---
 drivers/net/wireless/realtek/rtl8852bs/core/rtw_sta_mgt.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/net/wireless/realtek/rtl8852bs/core/rtw_sta_mgt.c b/drivers/net/wireless/realtek/rtl8852bs/core/rtw_sta_mgt.c
index 217543455b1e..1b2f791a6ca0 100644
--- a/drivers/net/wireless/realtek/rtl8852bs/core/rtw_sta_mgt.c
+++ b/drivers/net/wireless/realtek/rtl8852bs/core/rtw_sta_mgt.c
@@ -1335,7 +1335,8 @@ u32 rtw_init_self_stainfo(_adapter *padapter, enum phl_cmd_type cmd_type)
 				goto exit;
 			}
 		}
-		main_id = psta->phl_sta->macid;
+		if (psta->phl_sta)
+			main_id = psta->phl_sta->macid;
 	}
 	/* self mld and self stainfo, no need to link again */
 exit:
-- 
2.47.0

